src/hooks/useCheckMobileScreen.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from 'react';
+
+export const useCheckMobileScreen = (): boolean => {
+  const [width, setWidth] = useState<number>(window.innerWidth);
+  const handleWindowSizeChange = () => {
```

Comment:
Teraz tak, optymalizacja. Zauważ, że hook to właściwie funkcja, która wykona się za każdym razem, kiedy komponent, w którym jest ten hook się przerenderuje. Czyli jeśli np wrzucisz to do komponentu, gdzie mamy input, to ten hook się wykona za każdym razem, kiedy w inpucie wpiszemy nowy znak.
W linii 5 tworzymy nową funkcję. I ona jest tworzona nowa za każdym razem, kiedy hook jest wykonywany. Ale jako że `handleWindowSizeChange` jest używany tylko w useEffectie, więc cały ten handler można umieścić wewnątrz useEffect:
```
useEffect(() => {
  const handleWindowSizeChange = () => {
  ...

  window.addEventListener('resize', handleWindowSizeChange);
  return () => window.removeEventListener('resize', handleWindowSizeChange);
}, []);
```